internal/app/handlers/decode.go

```diff
@@ -0,0 +1,63 @@
+package handlers
+
+import (
+	"fmt"
+	"github.com/northmule/shorturl/config"
+	"github.com/northmule/shorturl/internal/app/services"
+	appStorage "github.com/northmule/shorturl/internal/app/storage"
+	"github.com/northmule/shorturl/internal/app/storage/models"
+	"io"
+	"net/http"
+	"regexp"
+)
+
+// DecodeHandler обработчик создания короткой ссылки
+func DecodeHandler(res http.ResponseWriter, req *http.Request) {
+	// Автотест не устанавливает нужный Content-Type https://github.com/Yandex-Practicum/go-autotests/blob/main/cmd/shortenertest/iteration1_test.go#L120
+	//if req.Header.Get("Content-Type") != "text/plain" {
+	//	http.Error(res, "expected Content-Type: text/plain", http.StatusBadRequest)
+	//	return
+	//}
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		http.Error(res, "error read body", http.StatusBadRequest)
+		return
+	}
```

Comment:
Не забывай о том, что нужно закрывать body ответа, после обработки ошибки!

```suggestion
	}
	defer body.Close()
```